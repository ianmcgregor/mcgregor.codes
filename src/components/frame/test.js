import React from 'react';
import {render} from 'enzyme';
import Frame from './';

const props = {
    src: 'https://www.example.com'
};

describe('Frame component', () => {
    it('should render', () => {
        const el = render((
            <Frame
                {...props}
            />
        ));
        expect(el).toMatchSnapshot();
    });
});
