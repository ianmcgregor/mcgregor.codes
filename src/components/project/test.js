import React from 'react';
import {render} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import {LinkStub} from '../../../test/helpers';
import Project from './';

const props = {
    type: 'ContentfulProject',
    __typename: 'ContentfulProject',
    id: 'c2OLhcgLiesoUqs48yMa0co',
    title: 'Google â€” Inside Abbey Road',
    year: 2015,
    image: {
        id: 'c3JDzQqutDa6uE6oOs22AEO',
        title: 'abbey road 10',
        description: '',
        resolutions: {
            aspectRatio: 1.7777777777777777,
            width: 400,
            height: 225,
            src: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&q=50',
            srcSet: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&h=225&q=50 1x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=600&h=338&q=50 1.5x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&h=450&q=50 2x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1200&h=675&q=50 3x',
            srcWebp: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&q=50&fm=webp',
            srcSetWebp: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&h=225&q=50&fm=webp 1x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=600&h=338&q=50&fm=webp 1.5x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&h=450&q=50&fm=webp 2x,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1200&h=675&q=50&fm=webp 3x'
        },
        file: {
            url: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png',
            fileName: 'abbey_road_10.png',
            contentType: 'image/png'
        },
        sizes: {
            aspectRatio: 1.7777777777777777,
            src: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&q=100',
            srcSet: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=200&h=113&q=100 200w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&h=225&q=100 400w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&h=450&q=100 800w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1200&h=675&q=100 1200w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1600&h=900&q=100 1600w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1920&h=1080&q=100 1920w',
            srcWebp: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&q=100&fm=webp',
            srcSetWebp: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=200&h=113&q=100&fm=webp 200w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&h=225&q=100&fm=webp 400w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=800&h=450&q=100&fm=webp 800w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1200&h=675&q=100&fm=webp 1200w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1600&h=900&q=100&fm=webp 1600w,\n//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=1920&h=1080&q=100&fm=webp 1920w',
            sizes: '(max-width: 800px) 100vw, 800px'
        },
        resize: {
            src: '//images.ctfassets.net/i8x1lkvuio7i/3JDzQqutDa6uE6oOs22AEO/9f09cad5e7504b8aeed7697d5c8832fa/abbey_road_10.png?w=400&fl=progressive&q=50',
            width: 400,
            height: 225,
            aspectRatio: 1.7777777777777777
        }
    },
    video: {
        id: 'c4JelMRzw1WCsoMGyKQkuUO',
        title: 'abbey road',
        description: '',
        file: {
            url: '//videos.ctfassets.net/i8x1lkvuio7i/4JelMRzw1WCsoMGyKQkuUO/c5c42ec2fd7a6b6b273e2e35b6f32649/abbey_road.mp4',
            fileName: 'abbey_road.mp4',
            contentType: 'video/x-m4v'
        }
    },
    text: {
        childMarkdownRemark: {
            html: '<p>I enjoyed pushing the Web Audio API to its limits on this virtual tour of Abbey Road studios which also entailed lots of WebGL and intricate UI work. I led the development over its year-long gestation. It was well received and won the <a href="https://thefwa.com/cases/inside-abbey-road">FWA Site of the Month.</a> and the Webby Award in the music category.</p>'
        }
    },
    link: 'https://insideabbeyroad.withgoogle.com/'
};

describe('Project component', () => {
    it('should render', () => {
        const el = render((
            <MemoryRouter>
                <Project
                    {...props}
                    LinkComponent={LinkStub}
                />
            </MemoryRouter>
        ));
        expect(el).toMatchSnapshot();
    });
});
