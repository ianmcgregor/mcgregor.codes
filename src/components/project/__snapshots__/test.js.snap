// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-jTzLTM eEjYel",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "/google_inside_abbey_road/",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-gzVnrw cjDudR",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-htoDjs dfKSvR",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "autoplay": "",
                      "class": "sc-gZMcBi jXcsyC",
                      "loop": "",
                      "muted": "",
                      "playsinline": "",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "src": "//videos.ctfassets.net/i8x1lkvuio7i/4JelMRzw1WCsoMGyKQkuUO/c5c42ec2fd7a6b6b273e2e35b6f32649/abbey_road.mp4",
                          "type": "video/x-m4v",
                        },
                        "children": Array [],
                        "name": "source",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "src": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "src": undefined,
                          "type": undefined,
                        },
                      },
                    ],
                    "name": "video",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autoplay": undefined,
                      "class": undefined,
                      "loop": undefined,
                      "muted": undefined,
                      "playsinline": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autoplay": undefined,
                      "class": undefined,
                      "loop": undefined,
                      "muted": undefined,
                      "playsinline": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-gqjmRU clbgIC",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "sc-bxivhb efuaUB",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-ifAKCX fwOaBU",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "sc-EHOje khSkhC",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "sc-bZQynM ixOPUe",
                          },
                          "children": Array [
                            Object {
                              "data": "Google â€” Inside Abbey Road",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-VigVT eDJikp",
                  "href": "https://insideabbeyroad.withgoogle.com/",
                  "rel": "nofollow",
                  "target": "_blank",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-bxivhb efuaUB",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "sc-fjdhpX gfwEQJ",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "sc-EHOje khSkhC",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "sc-bZQynM gyCMzi",
                                },
                                "children": Array [],
                                "name": "span",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                },
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                  "rel": undefined,
                  "target": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                  "rel": undefined,
                  "target": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "sc-VigVT eDJikp",
                "href": "https://insideabbeyroad.withgoogle.com/",
                "rel": "nofollow",
                "target": "_blank",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-bxivhb efuaUB",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "sc-fjdhpX gfwEQJ",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "sc-EHOje khSkhC",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "sc-bZQynM gyCMzi",
                              },
                              "children": Array [],
                              "name": "span",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-bxivhb efuaUB",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-ifAKCX fwOaBU",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "sc-EHOje khSkhC",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "sc-bZQynM ixOPUe",
                            },
                            "children": Array [
                              Object {
                                "data": "Google â€” Inside Abbey Road",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "h2",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
                "rel": undefined,
                "target": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
                "rel": undefined,
                "target": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": undefined,
        },
        "x-attribsPrefix": Object {
          "href": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-gqjmRU clbgIC",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-bxivhb efuaUB",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-ifAKCX fwOaBU",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-EHOje khSkhC",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "sc-bZQynM ixOPUe",
                        },
                        "children": Array [
                          Object {
                            "data": "Google â€” Inside Abbey Road",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-VigVT eDJikp",
                "href": "https://insideabbeyroad.withgoogle.com/",
                "rel": "nofollow",
                "target": "_blank",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-bxivhb efuaUB",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "sc-fjdhpX gfwEQJ",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "sc-EHOje khSkhC",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "sc-bZQynM gyCMzi",
                              },
                              "children": Array [],
                              "name": "span",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
                "rel": undefined,
                "target": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
                "rel": undefined,
                "target": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-VigVT eDJikp",
              "href": "https://insideabbeyroad.withgoogle.com/",
              "rel": "nofollow",
              "target": "_blank",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-bxivhb efuaUB",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-fjdhpX gfwEQJ",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "sc-EHOje khSkhC",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "sc-bZQynM gyCMzi",
                            },
                            "children": Array [],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-bxivhb efuaUB",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-ifAKCX fwOaBU",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "sc-EHOje khSkhC",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "sc-bZQynM ixOPUe",
                          },
                          "children": Array [
                            Object {
                              "data": "Google â€” Inside Abbey Road",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
              "rel": undefined,
              "target": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
              "rel": undefined,
              "target": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "href": "/google_inside_abbey_road/",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "sc-gzVnrw cjDudR",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-htoDjs dfKSvR",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autoplay": "",
                        "class": "sc-gZMcBi jXcsyC",
                        "loop": "",
                        "muted": "",
                        "playsinline": "",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "src": "//videos.ctfassets.net/i8x1lkvuio7i/4JelMRzw1WCsoMGyKQkuUO/c5c42ec2fd7a6b6b273e2e35b6f32649/abbey_road.mp4",
                            "type": "video/x-m4v",
                          },
                          "children": Array [],
                          "name": "source",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "src": undefined,
                            "type": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "src": undefined,
                            "type": undefined,
                          },
                        },
                      ],
                      "name": "video",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autoplay": undefined,
                        "class": undefined,
                        "loop": undefined,
                        "muted": undefined,
                        "playsinline": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autoplay": undefined,
                        "class": undefined,
                        "loop": undefined,
                        "muted": undefined,
                        "playsinline": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "href": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "article",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
